doc/js-spec/oic.md

```diff
@@ -0,0 +1,480 @@
+  attribute EventHandler<OicRequestEvent> onunobserverequest;
+  attribute EventHandler<OicRequestEvent> onretrieverequest;
+  attribute EventHandler<OicRequestEvent> ondeleterequest;
+  attribute EventHandler<OicResourceEvent>  onupdaterequest;
+  attribute EventHandler<OicResourceEvent>  oncreaterequest;
+
+  // update notification could be done automatically in most cases,
+  // but in a few cases manual notification is needed
+  Promise<void> notify(OicResourceInit resource);
+  // delete notifications should be made automatic by implementations
+
+  // enable/disable presence for this device
+  Promise<void> enablePresence(optional unsigned long long ttl);  // in ms
+  Promise<void> disablePresence();
+};
+
+interface OicRequestEvent : Event {
+  readonly attribute OicResourceId source;
+  readonly attribute OicResourceId target;
+
+  Promise<void> sendResponse(optional OicResource? resource);
+    // reuses request info (type, requestId, source, target) to construct response,
+    // sends back “ok”, plus the resource object if applicable
+
+  Promise<void> sendError(DOMString? name,
+                          optional DOMString? message,
+                          optional OicResourceInit resource);
+    // reuses request info (type, requestId, source, target) to construct response,
+};
+
+interface OicResourceEvent : OicRequestEvent {
+  readonly attribute OicResourceInit resource;
+};
+```
+
+### Error handling
+Errors are reported using ```OicError``` objects, which extend [Error](http://www.ecma-international.org/ecma-262/6.0/#sec-error-objects) with OIC specific information.
+```javascript
+interface OicError: Error {
+  readonly attribute USVString? deviceId;
+  readonly attribute OicResourceInit? resource;
+};
+```
+The error names map to [DOMError](https://heycam.github.io/webidl/#idl-DOMException-error-names). The following error names may be used when rejecting Promises: ```NotSupportedError```, ```SecurityError```, ```TimeoutError```, ```NotFoundError```, ```NoModificationAllowedError```, ```InvalidModificationError```, ```TypeMismatchError```, ```InvalidStateError```, ```InvalidAccessError```, ```InvalidNodeTypeError```, ```NotReadableError```, ```IndexSizeError```, ```DataCloneError```.
+
+The following error event is used for protocol errors:
+```javascript
+interface OicErrorEvent: Event {
+  OicError error;
+};
+```
+
+Code Examples
+-------------
+### Getting device configuration
+
+```javascript
+var oic = require('oic');
+if (oic.device.uuid) {  // configuration is valid
+  startServer();
+  startClient();
+} else {
+  console.log("Error: device is not configured.");
+}
+```
+
+### OIC Client contolling a remote red LED.
+```javascript
+// Discover a remote red light, start observing it, and make sure it's not too bright.
+var red = null;
+function startClient() {
+  // discover resources
+  oic.onresourcefound = function(event) {
+    if(event.resource && event.resource.id.path == "/light/ambience/red") {
```

Comment:
Are these paths different types? If not it should be three equals "===", otherwise it will get warnings